kv/memberlist/tcp_transport.go

```diff
@@ -426,7 +432,15 @@ func (t *TCPTransport) getAdvertisedAddr() string {
 func (t *TCPTransport) WriteTo(b []byte, addr string) (time.Time, error) {
 	t.sentPackets.Inc()
 	t.sentPacketsBytes.Add(float64(len(b)))
+	t.writeCh <- struct{}{}
```

Comment:
Is `WriteTo` only called from single goroutine? If not, it just became more blocking.

Reply:
I've added a test for this. Before my PR, calling `WriteTo` x times possibly takes `x * timeout delay`, following my PR, the calls are non-blocking